sourcegraph/sourcegraph.proto

```diff
@@ -414,8 +416,17 @@ message ReposCreateOp {
 
 	// CloneURL is the clone URL of the repository for mirrored
 	// repositories. If blank, a new hosted repository is created
-	// (i.e., a repo whose origin is on the server).
+	// (i.e., a repo whose origin is on the server). If Mirror is
+	// true, a clone URL must be provided.
 	string clone_url = 3 [(gogoproto.customname) = "CloneURL"];
```

Comment:
A non-mirror can still have a custom clone URL. Golang.org/x/tools and https://go.googlesource.com/tools for example. 

> On Jun 20, 2015, at 12:52, Dmitri Shuralyov <notifications@github.com> wrote:
> 
> In sourcegraph/sourcegraph.proto:
> 
> > @@ -414,8 +416,17 @@ message ReposCreateOp {
> >  
> >  	// CloneURL is the clone URL of the repository for mirrored
> >  	// repositories. If blank, a new hosted repository is created
> > -	// (i.e., a repo whose origin is on the server).
> > +	// (i.e., a repo whose origin is on the server). If Mirror is
> > +	// true, a clone URL must be provided.
> >  	string clone_url = 3 [(gogoproto.customname) = "CloneURL"];
> What happens if mirror is false and clone_url is still provided?
> 
> If that's meaningless, then mirror bool is unneeded since you could deduct if it's a mirror or not based on whether clone_url is blank or not.
> 
> —
> Reply to this email directly or view it on GitHub.
>